providers/spotify/spotify.go

```diff
@@ -0,0 +1,27 @@
+package dumpert
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+
+	"github.com/zquestz/s/providers"
+)
+
+func init() {
+	providers.AddProvider("spotify", &SpotifyProvider{})
+}
+
+// SpotifyProvider adheres to the Provider interface.
+type SpotifyProvider struct {
+}
+
+// BuildURI generates a search URL for spotify.
+func (p *SpotifyProvider) BuildURI(q string) string {
+	_, err := exec.LookPath("spotify")
```

Comment:
Seems Spotify.exe might not be in the default path. See https://superuser.com/questions/284342/what-are-path-and-other-environment-variables-and-how-can-i-set-or-use-them

I think we can just force windows to use the https:// search for now, and just support linux and OS X checks in the first pass of this provider.